### PR TITLE
Added TCP client docs from v5.0.9 to 20.10 and 21.2

### LINKED ISSUE #282: Document Compatibility mode for TCP Client in version 21.2

### LINKED ISSUE #283: Add 21.2 TCP client docs section
Waiting for the 21.2 TCP client release.

### PR DESCRIPTION
**Pull request by oskardudycz**
Copy/pasted TCP client docs from v5.0.9 to 20.10 and 21.2.
Updated compatibility mode docs for the 20.10 and 21.2 version.

Fixes #282 
Fixes #283

### CODE REVIEW

File: docs/clients/dotnet/20.10/connecting/compatibility-mode.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Compatibility Mode
+
+Compatibility Mode was added in v5.0.10 and v21.2.0 of the client to set certain configuration options depending on the Event Store version the client is connecting to. It allowa discovering whether the Event Store server is a v5 or v20 server.
```
Comment (hayley-jean) [inline docs/clients/dotnet/20.10/connecting/compatibility-mode.md]:
```suggestion
Compatibility Mode was added in v5.0.10 and v21.2.0 of the client to set certain configuration options depending on the Event Store version the client is connecting to. It allows discovering whether the Event Store server is a v5 or v20 server.
```

File: docs/clients/dotnet/20.10/embedded/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+# Embedded client
```
Comment (hayley-jean) [inline docs/clients/dotnet/20.10/embedded/README.md]: We haven't released the embedded client for v20.10 and above, should we remove the docs relating to it?

File: docs/clients/dotnet/21.2/connecting/compatibility-mode.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Compatibility Mode
+
+Compatibility Mode was added in v21.2 of the client to set certain configuration options depending on the Event Store version the client is connecting to.
+
+At the moment, the v21.2 client supports two modes:
+
+- "5" to allow connection to Event Store servers v5.0.10.
```
Comment (hayley-jean) [inline docs/clients/dotnet/21.2/connecting/compatibility-mode.md]:
```suggestion
- "5" to allow connection to v5 Event Store servers.
```
Auto-compatibility is compatible with any version of v5

File: docs/clients/dotnet/21.2/connecting/compatibility-mode.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+</xode-block>
+<xode-block title="Using Gossip Seeds">
+
+<<< @/docs/clients/dotnet/21.2/sample-code/DotNetClient/CompatibilityMode.cs#AutoCompatibilityWithGossipSeeds
+</xode-block>
+</xode-group>
+
+## v5 Compatibility Mode
+
+The v5 Compatibility Mode allows the v21.2 client to connect to a cluster running in ES Cloud.
```
Comment (hayley-jean) [inline docs/clients/dotnet/21.2/connecting/compatibility-mode.md]:
```suggestion
The v5 Compatibility Mode allows the v21.2 client to connect to a v5 cluster.
```

File: docs/clients/dotnet/21.2/connecting/compatibility-mode.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+You can set this with `CompatibilityMode=5` in the connection string, or with `.SetCompatibilityMode("5")` in the connection settings.
+
+For example:
+
+```csharp
+var connectionString = $"ConnectTo=discover://{CloudClusterId}.mesdb.eventstore.cloud:2113?UseSslConnection=true;ValidateServer=true;CompatibilityMode=5;"
```
Comment (hayley-jean) [inline docs/clients/dotnet/21.2/connecting/compatibility-mode.md]:
```suggestion
var connectionString = $"ConnectTo=discover://{cluster_dns}:2113?TargetHost={cluster_dns};CompatibilityMode=5;"
```

File: docs/clients/dotnet/21.2/embedded/README.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
+# Embedded client
```
Comment (hayley-jean) [inline docs/clients/dotnet/21.2/embedded/README.md]: See previous comment about embedded client

**Review by hayley-jean — CHANGES_REQUESTED** [review]

File: docs/clients/dotnet/21.2/embedded/README.md (hunk 6 above)
Comment (oskardudycz) [reply docs/clients/dotnet/21.2/embedded/README.md]: Nice catch - removed in https://github.com/EventStore/documentation/pull/288/commits/ba3f7e7c27574e7c876a5ae749868472ce656f95.

File: docs/clients/dotnet/20.10/embedded/README.md (hunk 2 above)
Comment (oskardudycz) [reply docs/clients/dotnet/20.10/embedded/README.md]: Nice catch - removed in https://github.com/EventStore/documentation/pull/288/commits/ba3f7e7c27574e7c876a5ae749868472ce656f95.

File: docs/clients/dotnet/21.2/connecting/compatibility-mode.md (hunk 5 above)
Comment (oskardudycz) [reply docs/clients/dotnet/21.2/connecting/compatibility-mode.md]: Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

File: docs/clients/dotnet/21.2/connecting/compatibility-mode.md (hunk 4 above)
Comment (oskardudycz) [reply docs/clients/dotnet/21.2/connecting/compatibility-mode.md]: Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

File: docs/clients/dotnet/21.2/connecting/compatibility-mode.md (hunk 3 above)
Comment (oskardudycz) [reply docs/clients/dotnet/21.2/connecting/compatibility-mode.md]: Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

File: docs/clients/dotnet/20.10/connecting/compatibility-mode.md (hunk 1 above)
Comment (oskardudycz) [reply docs/clients/dotnet/20.10/connecting/compatibility-mode.md]: Fixed in https://github.com/EventStore/documentation/pull/288/commits/151543c1acbb559165d3d983279156be408f5eb2.

**Comment by oskardudycz** [comment]
@hayley-jean @alexeyzimarev I think that I applied all of your comments

**Review by alexeyzimarev — APPROVED** [review]

**Review by hayley-jean — APPROVED** [review]

**Pull request merged**